adls/include/adls_client.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#ifdef _WIN32
+#ifdef azure_storage_adls_EXPORTS
+#define AZURE_STORAGE_ADLS_API __declspec(dllexport)
+#else
+#define AZURE_STORAGE_ADLS_API __declspec(dllimport)
+#endif
+#else /* ifdef _WIN32 */
+#define AZURE_STORAGE_ADLS_API
+#endif
+
+#include "blob/blob_client.h"
+#include "nlohmann_json_parser.h"
+
+namespace azure { namespace storage_adls {
+    using storage_account = azure::storage_lite::storage_account;
+    using executor_context = azure::storage_lite::executor_context;
+
+    struct list_filesystems_item
+    {
+        std::string name;
+    };
+
+    struct list_filesystems_result
+    {
+        std::vector<list_filesystems_item> filesystems;
+        std::string continuation_token;
+    };
+
+    class adls_client final
+    {
+    public:
+        adls_client(std::shared_ptr<storage_account> account, int max_concurrency, bool exception_enabled = true) : m_account(account), m_blob_client(std::make_shared<azure::storage_lite::blob_client>(account, max_concurrency)), m_context(m_blob_client->context()), m_exception_enabled(exception_enabled)
+        {
+            m_context->set_json_parser(std::make_shared<nlohmann_json_parser>());
+        }
```

Comment:
If the constructor is in the .cpp file, then user won't be including 3rd party dependency in their own binaries.

Reply:
Sorry, my bad, `nlohmann_json_parser` is actually internal .h file. However I still think it is worth the change.

---
In reply to: [351631571](https://github.com/Azure/azure-storage-cpplite/pull/50#discussion_r351631571) [](ancestors = 351631571)

Reply:
ok

Reply:
fixed in https://github.com/Azure/azure-storage-cpplite/pull/50/commits/829c47222865f8be7f2524a745c3f31cfa1c968b